libbeat/logp/metrics.go

```diff
@@ -0,0 +1,186 @@
+package logp
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/elastic/beats/libbeat/monitoring"
+)
+
+type snapshotVisitor struct {
+	snapshot snapshot
+	level    []string
+}
+
+type snapshot struct {
+	bools   map[string]bool
+	ints    map[string]int64
+	floats  map[string]float64
+	strings map[string]string
+}
+
+// logExpvars logs at Info level the integer expvars that have changed in the
+// last interval. For each expvar, the delta from the beginning of the interval
+// is logged.
+func logMetrics(metricsCfg *LoggingMetricsConfig) {
+	if metricsCfg.Enabled != nil && *metricsCfg.Enabled == false {
```

Comment:
yeah, I just moved the old code over. Publishing will be addressed in another PR.